File: model/src/main/java/aop/profiling/DoNotProfile.java
package aop.profiling;

import static java.lang.annotation.ElementType.METHOD;
import static java.lang.annotation.ElementType.TYPE;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

/**
 * @author Christopher Zhong
 * @since 7.0.0
 */
@Retention(RUNTIME)
@Target({ TYPE, METHOD })
public @interface DoNotProfile {}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/aop/profiling/DoNotProfile.kt
package runtimemodels.chazm.model.aop.profiling


/**
 * The [DoNotProfile] annotation can be used to inform any profiling interceptors to ignore the annotated class or method.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class DoNotProfile
